sdcm/cluster_k8s/__init__.py

```diff
@@ -1703,9 +1704,39 @@ def manage_file_in_scylla_config_map(self, filename: str, namespace: str = SCYLL
                 scylla_config_map[filename] = new_data_as_str
             self.scylla_restart_required = True
 
-    def remote_scylla_yaml(self, namespace: str = SCYLLA_NAMESPACE) -> ContextManager:
-        return self.manage_file_in_scylla_config_map(
-            filename='scylla.yaml', namespace=namespace)
+    @contextlib.contextmanager
+    def remote_scylla_yaml(self, namespace: str = SCYLLA_NAMESPACE) -> ContextManager[ScyllaYaml]:
```

Comment:
I suspect it was not tested because scylla operator handles lots of scylla yaml parameters itself.
And it will redefine stuff.

So, the proper solution here, IMHO, would be an implementation of a proper interface of that ScyllaYaml class for K8S backends - just make attrs work not requiring dict-like handling as it is now.

Reply:
see:
```
            # Create ScyllaYaml object with no defaults (model_construct bypasses validation and defaults)
            new_scylla_yaml = ScyllaYaml.model_construct(**old_data)
```